src/Schema/Index/IndexedColumn.php

```diff
@@ -15,13 +16,17 @@ final class IndexedColumn
      */
     public function __construct(private readonly UnqualifiedName $columnName, private readonly ?int $length)
     {
+        if ($length !== null && $length <= 0) {
+            throw InvalidIndexDefinition::fromNonPositiveColumnLength($length);
+        }
```

Comment:
This is not a breaking change. `Index` will catch this exception and turn into a deprecation notice. The exception will be thrown only upon an attempt to access the indexed columns via the new API: https://github.com/doctrine/dbal/blob/feed861ebadf6a3f1b0723b911c6848be35b5611/tests/Schema/IndexTest.php#L255-L264